### PR TITLE
Add AzureOpenAI support

### PR DESCRIPTION
**Pull request by antonum**
This pull request adds Azure OpenAI support to the demo.

In addition to all the standard OpenAI parameters it requires:
```
OPENAI_API_TYPE=azure
OPENAI_API_VERSION=2022-12-01
OPENAI_API_BASE=https://YOUR_ENDPOINT.openai.azure.com/
OPENAI_COMPLETIONS_ENGINE=text-davinci-003
```

Current service limits for Azure OpenAI S0 make Embedding API unusable for the purposes of this demo. Instead, if `OPENAI_API_TYPE=azure` is detected - the demo would load and use HugginFace HuggingFace `all-MiniLM-L6-v2` embedding, which can be considerably slower then remote OpenAI embeddings.

### CODE REVIEW

File: app/qna/llm.py [hunk 1]
```diff
@@ -11,19 +11,27 @@
...
 
 def create_vectorstore() -> Redis:
     """Create the Redis vectorstore."""
     import pandas as pd
 
-    from langchain.embeddings import OpenAIEmbeddings
 
-     # Init OpenAI Embeddings
-    oai_embeddings = OpenAIEmbeddings()
+    if OPENAI_API_TYPE=="azure":
```
Comment (tylerhutcherson) [inline app/qna/llm.py:18]: Can we pull this into a helper function like this?

`get_embeddings() -> Embddings`

I will need to reuse this for the semantic cache implementation coming soon on another branch I have

File: app/qna/llm.py [hunk 2]
```diff
@@ -98,12 +106,24 @@ def make_qna_chain():
...
     chain_type_kwargs = {"prompt": prompt}
 
-    chain = RetrievalQA.from_chain_type(
-        llm=OpenAI(),
-        chain_type="stuff",
-        retriever=redis.as_retriever(),
-        return_source_documents=True,
-        chain_type_kwargs=chain_type_kwargs
-    )
+    if OPENAI_API_TYPE=="azure":
```
Comment (tylerhutcherson) [inline app/qna/llm.py]: Can we pull this into a function like this:

`get_llm() -> LLM:`

and return the appropriate LLM based on the env vars?

Then just pass the llm object into the RetrievalQA chain

**Review by tylerhutcherson — CHANGES_REQUESTED** [review]
Thanks Anton! Two small changes requested to set us up for future growth, but this is awesome!!

File: app/qna/llm.py (hunk 1 above)
Comment (antonum) [reply app/qna/llm.py:18]: done.

File: app/qna/llm.py (hunk 2 above)
Comment (antonum) [reply app/qna/llm.py]: done

**Review by tylerhutcherson — APPROVED** [review]

**Pull request merged**